### PR TITLE
book: Add chapter about FlaggedStorage

### PR DESCRIPTION
**Pull request by Rhuagh**
This will need rebasing once #395 is merged.

---
This change is [Reviewable](https://reviewable.io/reviews/slide-rs/specs/397)

### CODE REVIEW

File: book/src/12_tracked.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# `FlaggedStorage` and modification events
+
+In most games you will have many entities, but from frame to frame only a few of them will 
+actually need to have components updated.
```
Comment (WaDelma) [inline book/src/12_tracked.md]: For me it's bit unclear what updating components means here: Maybe add some examples on it.
You could also mention keeping some external data structure on sync with changes.

File: book/src/12_tracked.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+impl<'a> System<'a> for Sys {
+    type SystemData = (
+        ReadStorage<'a, Data>,
+        WriteStorage<'a, SomeOtherData>,
+    );
+    
+    fn run(&mut self, (data, some_other_data): Self::SystemData) {
+        self.dirty.clear();
+        data.populate_inserted(&mut self.insert_reader_id.as_mut().unwrap(), &mut self.dirty);
+        data.populate_inserted(&mut self.modify_reader_id.as_mut().unwrap(), &mut self.dirty);
```
Comment (WaDelma) [inline book/src/12_tracked.md]: Maybe add comment that you can have multiple different bitsets for different events.

**Comment by Rhuagh** [comment]
Fixed.

**Comment by Rhuagh** [comment]
I will fix the history once the other PR is merged.

**Comment by GraemeWilde** [comment]
*[book/src/12_tracked.md, line 49 at r3](https://reviewable.io/reviews/slide-rs/specs/397#-LC0ppE74Cy3NcktzW-m:-LC0ppE74Cy3NcktzW-n:b-hmga0w) ([raw file](https://github.com/slide-rs/specs/blob/24e7ffcf4f957b687333262e8e1371cc225e145a/book/src/12_tracked.md#L49)):*
> ```Markdown
>         
>         // Note that we could use separate bitsets here, we only use one to simplify the example
>         data.populate_inserted(&mut self.modify_reader_id.as_mut().unwrap(), &mut self.dirty);
> ```

I think you forgot to change this line to populate_modified

---


*Comments from [Reviewable](https://reviewable.io/reviews/slide-rs/specs/397)*

File: book/src/11_advanced_component.md [hunk 3]
```diff
@@ -55,7 +55,7 @@ impl<'a> System<'a> for Sys {
 
 Using `NullStorage` is recommended for marker components, since they don't contain
 any data and as such will not consume any memory. This means we can represent them using 
-only a bitset. Note that `NullStorage` will only work for components that are ZST (i.e. a
```
Comment (torkleyy) [inline book/src/11_advanced_component.md]: err I think you removed this accidentally

File: book/src/11_advanced_component.md [hunk 4]
```diff
@@ -124,8 +124,8 @@ impl<'a> System<'a> for Render {
     );
     
     fn run(&mut self, (active_cam, camera, transform, mesh) : Self::SystemData) {
-        let camera = camera.get(active_cam.0).unwrap();
-        let view_matrix = transform.get(active_cam.0).unwrap().invert();
```
Comment (torkleyy) [inline book/src/11_advanced_component.md]: I don't understand this change either.

Comment (Rhuagh) [reply book/src/11_advanced_component.md]: Umm, that must be rebase doing something wonky, hold on.

**Comment by Rhuagh** [comment]
There, now.

**Comment by Rhuagh** [comment]
Will push all these in a single PR.

**Pull request closed without merging**